hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/oauth2/ClientCredsTokenProvider.java

```diff
@@ -38,19 +42,27 @@
 
   private static final Logger LOG = LoggerFactory.getLogger(AccessTokenProvider.class);
 
-
   public ClientCredsTokenProvider(final String authEndpoint,
-                                  final String clientId, final String clientSecret) {
-
-    Preconditions.checkNotNull(authEndpoint, "authEndpoint");
-    Preconditions.checkNotNull(clientId, "clientId");
-    Preconditions.checkNotNull(clientSecret, "clientSecret");
+      final String clientId, final String clientSecret)
+      throws AzureBlobFileSystemException {
+    validateClientCredsTokenProvider(
```

Comment:
That is not going to work. You probably want to keep non-null check here, since this class does not depend on Hadoop configuration.